libstorage/storage.c

```diff
@@ -181,10 +180,12 @@ static void storage_reqthr(void *arg)
 			mutexUnlock(ctx->lock);
 		}
 		else if (ctx->state == state_run) {
```

Comment:
You could just unlock mutex after handling all cases (similar to line 169).

Reply:
Done